lib/Serialization/Serialization.cpp

```diff
@@ -4615,6 +4796,25 @@ void Serializer::writeAST(ModuleOrSourceFile DC,
       index_block::EntryPointLayout EntryPoint(Out);
       EntryPoint.emit(ScratchRecord, entryPointClassID.getValue());
     }
+
+    if (enableDeclMemberNamesTable) {
```

Comment:
Idea that might help with development: you could *always* write out the `DECL_MEMBER_TABLES_BLOCK_ID`, but only actually use it when `options.EnableDeclMemberNamesTable` is `true`. That way, your normal build of the standard library, overlays, etc. has all of the information you need while you're testing out the compiler, and it becomes much easier to toggle the flag to try this code path out on existing tests/projects.

Reply:
I'm still a bit shy about turning it on that much yet -- it could still explode a fair number of things by making gigantic tables or corrupting the output or something -- but probably will do in the next iteration of testing :)

Reply:
Ok